src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncerTest.java

```diff
@@ -564,13 +591,16 @@ public final void testCheckAndCreateLeasesForNewShardsAtTimestampAndClosedShard(
     public final void testCheckAndCreateLeasesForNewShardsAtTimestampAndClosedShardWithDeleteLeaseExceptions()
             throws KinesisClientLibIOException, DependencyException, InvalidStateException, ProvisionedThroughputException,
             IOException {
+        final Set<String> expectedLeaseKeysToCreate = new HashSet<>(Arrays.asList(
+                "shardId-8", "shardId-4", "shardId-9", "shardId-10"
+        ));
```

Comment:
Shouldn't AT_TIMESTAMP create leases the same as TRIM? I.e. 0, 1, 2, 3, 4, 5 ?

Reply:
* 0 1 2 3 4   5- shards till epoch 102
     * \ / \ / |   |
     *  6   7  4   5- shards from epoch 103 - 205
     *   \ /   |  /\
     *    8    4 9  10 - shards from epoch 206 (open - no ending sequenceNumber)

At timestamp is at 1000, so I think this is the correct behavior (i.e should be same as TIP).

Reply:
Got it, if this test is using empty lease table synchronizer then yeah that's the correct behavior. Maybe we can call that out in a comment.